### PR TITLE
Ut 3033/upgrade to 2020

### PR DESCRIPTION
**Pull request by BadSingleton**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +46,10 @@ endif()
 # don't generate stubs for inherited virtuals e.g. GetSelected
 SET(CMAKE_SWIG_FLAGS -fvirtual)
 
+if (${FBXSDK_VERSION} STREQUAL "")
+    set(FBXSDK_VERSION "2020.0")
+endif()
```
Comment (benoithudson) [inline CMakeLists.txt]: This should be only in the context of stevedore.

FindFBXSDK will pick the latest automatically.

File: CMakeLists.txt [hunk 2]
```diff
@@ -56,21 +60,33 @@ if (${USE_STEVEDORE} STREQUAL "ON")
...
         set(swig_artifact_name swig-win-x64)
         set(swig_artifact_id 3.0.12_2e184e8ce59cca5da026ac275ef53aff60e8b60ec5edb576efced3f7168f82d8.7z)
         set(swig_executable_name swig.exe)
     elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
         set(fbxsdk_artifact_name fbxsdk-linux-x64)
-        set(fbxsdk_artifact_id 2018.1.1_9fd6bb314e0761415879fe91228f536902aa90082adb198889f71bdf094415fd.7z)
+        if(${FBXSDK_VERSION} STREQUAL "2020.0")
+            set(fbxsdk_artifact_id 2020.0_008abd4dfed259df5d8bbf2789ed3fc8b7b496f7b97668e940d42ec9eb2652ae.7z)
+        else()
+            set(fbxsdk_artifact_id 2018.1.1_9fd6bb314e0761415879fe91228f536902aa90082adb198889f71bdf094415fd.7z)
```
Comment (benoithudson) [inline CMakeLists.txt]: Do we ever want to build (on stevedore) with 2018 ever again?

File: cmake/FindFBXSDK.cmake [hunk 3]
```diff
@@ -103,6 +103,11 @@ if (FBXSDK_VERSION VERSION_GREATER 2019.1)
     list(APPEND FBXSDK_LIBRARY "-lxml2" "-lz")
   endif()
 endif()
+if(FBXSDK_VERSION VERSION_GREATER_EQUAL 2020.0)
+  if(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
+    list(APPEND FBXSDK_LIBRARY "-liconv")
```
Comment (benoithudson) [inline cmake/FindFBXSDK.cmake]: can you put this in with the `if(APPLE)` block above? iconv is already being added in one circumstance.

File: tests/RuntimeTests/Editor/UnitTests/FbxGeometryBaseTest.cs [hunk 4]
```diff
@@ -38,7 +38,8 @@ override public void TestBasics(T geometryBase, FbxNodeAttribute.EType typ)
             // It doesn't crash getting negative nor past-the-end.
             // The vector returned is documented to be (0,0,0,1) but actually
             // seems to be (0,0,0,epsilon).
-            geometryBase.GetControlPointAt(-1);
+            UnityEngine.Debug.Log("Uncomment me!");
+            // geometryBase.GetControlPointAt(-1);
```
Comment (benoithudson) [inline tests/RuntimeTests/Editor/UnitTests/FbxGeometryBaseTest.cs]: ?

File: CMakeLists.txt (hunk 1 above)
Comment (benoithudson) [reply CMakeLists.txt]: Actually I'm not sure we should ever support this.

File: tests/RuntimeTests/Editor/UnitTests/FbxGeometryBaseTest.cs (hunk 4 above)
Comment (BadSingleton) [reply tests/RuntimeTests/Editor/UnitTests/FbxGeometryBaseTest.cs]: upgrading to 2020 causes a crash here. UT-2477 addresses the issue. the fix is done locally, waiting for this to go in to make the PR

Comment (BadSingleton) [reply tests/RuntimeTests/Editor/UnitTests/FbxGeometryBaseTest.cs]: upgrading to 2020 causes a crash here. UT-2477 addresses the issue. the fix is done locally, waiting for this to go in to make the PR

File: cmake/FindFBXSDK.cmake (hunk 3 above)
Comment (BadSingleton) [reply cmake/FindFBXSDK.cmake]: I can

File: CMakeLists.txt (hunk 2 above)
Comment (BadSingleton) [reply CMakeLists.txt]: I do not know

File: cmake/FindFBXSDK.cmake (hunk 3 above)
Comment (BadSingleton) [reply cmake/FindFBXSDK.cmake]: I can

Comment (BadSingleton) [reply cmake/FindFBXSDK.cmake]: by the by, will we ever be using USE_CUSTOM_FBXSDK?

File: .yamato/yamato.yml [hunk 5]
```diff
@@ -72,7 +72,7 @@ build_win:
     flavor: {{ win_platform.flavor}}
   commands:
     - choco -v -y install 7zip
-    - python build.py --stevedore --verbose --clean --yamato
+    - python build.py --stevedore --verbose --clean --fbxsdkversion 2020.0  --yamato
```
Comment (benoithudson) [inline .yamato/yamato.yml]: I think you can nuke this?

File: cmake/FindFBXSDK.cmake [hunk 6]
```diff
@@ -84,26 +84,22 @@ endif()
...
-        list(APPEND FBXSDK_LIBRARY "-liconv")
-      endif()
+      list(APPEND FBXSDK_LIBRARY ${COCOA_LIBRARY} "-liconv")
   endif()
 endif()
 
-if (FBXSDK_VERSION VERSION_GREATER 2019.1)
-# since 2019.1 we ahve to explicitly link against libxml and zlib. Fortunately 
+# since 2019.1 we have to explicitly link against libxml and zlib. Fortunately 
 # for us, FBS SDk ships them.
```
Comment (benoithudson) [inline cmake/FindFBXSDK.cmake]: `FBS SDk`

File: tests/RuntimeTests/Editor/UnitTests/FbxGeometryBaseTest.cs (hunk 4 above)
Comment (benoithudson) [reply tests/RuntimeTests/Editor/UnitTests/FbxGeometryBaseTest.cs]: Oh fun. So, this "uncomment me" line is going away in the next PR?

**Pull request merged**